server/controllers/mapsController.js

```diff
@@ -0,0 +1,55 @@
+const axios = require("axios");
+
+const KEY = process.env.GOOGLE_MAPS_API_KEY;
+const urlHead = "https://maps.googleapis.com/maps/api"
+
+const getAutoCompletePredictions = async (input) => {
+    const url =
+        "https://maps.googleapis.com/maps/api/place/autocomplete/json" +
+        `?input=${input}&type=address&location=43.653225,-79.383186&radius=5000000&key=${KEY}`;
+
+    const { data } = await axios.get(url);
+    return data.predictions.map(({ description }) => description);
+};
+
+const getStaticMapImage = async (center) => {
+    const url =
+        "https://maps.googleapis.com/maps/api/staticmap" +
+        `?center=${center}&scale=2&zoom=13&size=600x300&maptype=roadmap&key=${KEY}`;
+
+    return await axios({ method: "get", url, responseType: "stream" });
+};
+
+const getLocationCoordinates = async (formattedAddress) => {
+    const url =
+        "https://maps.googleapis.com/maps/api/geocode/json" +
+        `?address=${encodeURIComponent(formattedAddress)}&key=${KEY}`;
+
+    const { data: { status, results } } = await axios.get(url);
+
+    if (status !== 'OK') {
+        console.error(status)
+        throw Error()
+    }
+
+    let lat, lng;
+    if (
+        results && Array.isArray(results) &&
+        results[0] && typeof results[0] === "object"
```

Comment:
Are all of these checks necessary? seems like a lot

Reply:
We can actually get by without using `results && Array.isArray(results)` as we are already checking `status` earlier. I'll amend the change.